### ISSUE TITLE
Linting Rule: Invalid EngineVersion on ElastiCache cluster

### ISSUE THREAD

**Issue by DeflateAwning** [aws-cloudformation/cfn-lint#3820]
### Is this feature request related to a new rule or cfn-lint capabilities?

rules

### Describe the feature you'd like to request

I want the invalid redis version (i.e., the patch version is specified) to be added as a rule:

```yaml
  RedisCluster:
    Type: AWS::ElastiCache::CacheCluster
    # Deletion is the default policy.
    DeletionPolicy: Delete
    UpdateReplacePolicy: Delete
    Properties:
      Engine: redis
      CacheNodeType: cache.t4g.micro
      # RELATED TO ISSUE: Do not include patch versions in the EngineVersion (because it's invalid).
      EngineVersion: 7.1 # 7.1.0 is invalid.
      NumCacheNodes: 1
      ClusterName: "cluster-name"
      VpcSecurityGroupIds:
        - !Ref RedisSecurityGroup
      CacheSubnetGroupName: !Ref RedisSubnetGroup
```

### Describe the solution you'd like

The code snippet above should induce a linting error.

### Additional context

Just wasted 10 minutes on a deployment because of this. 

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
